packages/opentelemetry-plugin-dns/src/utils.ts

```diff
@@ -0,0 +1,211 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Span, CanonicalCode, Status, Attributes } from '@opentelemetry/types';
+import { AttributeNames } from './enums/AttributeNames';
+import { AddressFamily } from './enums/AddressFamily';
+import * as dns from 'dns';
+import { IgnoreMatcher } from './types';
+
+/**
+ * Set error attributes on the span passed in params
+ * @param err the error that we use for filling the attributes
+ * @param span the span to be set
+ * @param nodeVersion the node version
+ */
+export const setError = (
+  err: NodeJS.ErrnoException,
+  span: Span,
+  nodeVersion: string
+) => {
+  const { code, message, name } = err;
+  const attributes = {
+    [AttributeNames.DNS_ERROR_MESSAGE]: message,
```

Comment:
is message a limited set of of possible values or unlimited (e.g. has some unique identifier in it or something)? If it's the latter it should probably go onto the span as an event

Reply:
I believe we have a limited set. I'm also not sure if people will use this tag for filtering (Pretty sure, they won't for Jaeger since you need to have an exact match). Perhaps we could create an another PR since we could change also http/https plugins as well if we want to create an event instead.